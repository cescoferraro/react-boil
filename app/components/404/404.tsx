import * as React from "react"

export const NoMatchContainer = (props) => {
    return (
        <div>
            <h2>NOT FOUND!</h2>
        </div>
    )
}
