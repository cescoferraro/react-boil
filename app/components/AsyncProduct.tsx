import * as React from "react"
import { asyncComponent } from 'react-async-component'

export default asyncComponent({
    resolve: () => Promise.resolve().then(() => require("./Product.tsx"))
});
