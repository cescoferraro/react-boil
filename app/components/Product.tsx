import * as React from "react"
declare var System: any
import * as CSS from "../css/teste.css"

export default () => (
    <div className={CSS.test}>
        <h2>I am on a separate Bundle!</h2>
    </div>
)
