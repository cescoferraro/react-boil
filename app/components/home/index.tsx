import * as React from "react"
import * as CSS from "./css/home.css"
const test = require("../../../README.org")

export const HomeComponent = ({ }) => {
    return <div className={CSS.container} dangerouslySetInnerHTML={{ __html: test }} />
}
