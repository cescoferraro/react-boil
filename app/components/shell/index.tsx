import { ListSubheader as Subheader } from 'material-ui/List'
import * as cs from "classnames"
import * as React from "react"
import * as CSS from "./shell.css"
import Drawer from "material-ui/Drawer"
import Menu, { MenuItem } from 'material-ui/Menu';
const BoilLogo = require("../../../shared/images/boil.svg")
const Cancel = require("../../../shared/images/cancel.svg")
const Boy = require("../../../shared/images/boy.svg")
const Girl = require("../../../shared/images/girl.svg")
import ReduxToastr from "react-redux-toastr"
import { withStyles, createStyleSheet } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import Button from 'material-ui/Button';
import IconButton from 'material-ui/IconButton';
import MenuIcon from 'material-ui-icons/Menu';

const styleSheet = createStyleSheet({
    root: {
        width: '100%',
    },
    flex: {
        flex: 1,
    },
});



export const Shell = withStyles(styleSheet)((props) => {
    const closeOnClick = (open) => { props.DRAWER_ACTION(open) }
    const boilLogo = (
        <BoilLogo
            onClick={props.DRAWER_TOGGLE_ACTION}
            className={CSS.button}
        />
    )
    const goBack = () => {
        props.GO_HOME_ACTION()
        props.DRAWER_ACTION(false)
    }
    const goUser = (id) => () => {
        props.USER_ACTION(id)
        props.DRAWER_ACTION(false)
    }
    return (
        <div>
            <div className={props.classes.root}>
                <AppBar position="static">
                    <Toolbar>
                        <IconButton color="contrast" aria-label="Menu">
                            <MenuIcon />
                        </IconButton>
                        <Typography
                            onClick={goBack}
                            type="title" color="inherit" className={props.classes.flex}>
                            React-boil
			</Typography>

                        <BoilLogo
                            onClick={props.DRAWER_TOGGLE_ACTION}
                            className={CSS.button}
                        />
                    </Toolbar>
                </AppBar>
            </div>
            <div className={CSS.container}>
                {props.children}
            </div>
            <ReduxToastr
                timeOut={4000}
                newestOnTop={false}
                preventDuplicates={true}
                position="top-left"
                transitionIn="fadeIn"
                transitionOut="fadeOut"
                progressBar={true}
            />
            <Drawer
                docked={false}
                open={props.drawer}
            >
                <div className={cs(CSS.flex, CSS.main)}>
                    <BoilLogo
                        onClick={goBack}
                        className={cs(CSS.svg)}
                    />
                </div>
                <Subheader>React-boil</Subheader>
                <MenuItem
                    onClick={goUser(432)}
                >
                    User 123
                </MenuItem>
                <MenuItem
                    onClick={goUser(324)}
                >
                    User 456
                </MenuItem>
                <MenuItem
                    role="ksdjnf"
                    onClick={props.DRAWER_TOGGLE_ACTION}
                >
                    Close
                </MenuItem>
            </Drawer>
        </div>
    )
})
