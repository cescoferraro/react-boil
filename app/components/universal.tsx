import universal from "react-universal-component"
declare var System: any

export const UniversalComponent = universal(
    () => System.import(/* webpackChunkName: 'Product' */ "./product"),
    {
        resolve: () => require.resolveWeak("./product"),
        chunkName: "Product",
        onError: (erros) => {
            console.log(erros)
        }
    }
)
