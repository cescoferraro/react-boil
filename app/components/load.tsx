import universal from "react-universal-component"

export const UniversalComponent = universal(() => import("./product"), {
    resolve: () => require.resolveWeak("./product"),
    chunkName: "Universla",
    onError: (erros) => {
        console.log(erros)
    }
})
