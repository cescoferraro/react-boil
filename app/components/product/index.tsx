import universal from "react-universal-component"
declare var System: any

export const UniversalComponent = universal(
    () => System.import(/* webpackChunkName: 'Product' */ "./Product"),
    {
        resolve: () => require.resolveWeak("./Product"),
        chunkName: "Product"
    }
)
