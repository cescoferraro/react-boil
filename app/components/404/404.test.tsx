import * as React from "react"
import { NoMatchContainer } from "./404"
import { profileStartup } from "../../../store/reducers"
import { MUIProvider } from "../../../shared/components/MUIProvider"

describe("<NoMatchContainer />", () => {
    it('404', () => {
        const component = mount(
            <MUIProvider>
                <NoMatchContainer />
            </MUIProvider>
        )
        expect(component.text()).toContain('NOT FOUND!');
        expect(component).toMatchSnapshot()
    })
})
