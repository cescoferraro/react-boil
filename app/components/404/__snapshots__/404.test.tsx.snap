// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoMatchContainer /> 404 1`] = `
<Component>
  <MuiThemeProvider>
    <Component>
      <div>
        <h2>
          NOT FOUND!
        </h2>
      </div>
    </Component>
  </MuiThemeProvider>
</Component>
`;
